packages/react-centra-checkout/src/PaymentEmbed/index.tsx

```diff
@@ -75,8 +90,11 @@ const PaymentEmbed = memo(function PaymentEmbed(
         })
         .catch((err: unknown) => {
           console.error('@noaignite/react-centra-checkout: Could not submit payment')
-          // @ts-expect-error -- TODO: Fix this
-          setPaymentCallbackError(err)
+
+          if (isRecordStringString(err)) {
```

Comment:
The reason were using `isRecordStringString` function, instead of a `isCheckoutApiError` is to avoid false positives if the `CheckoutApi.Error` type would be changed in the future. 
The `CheckoutApi.Error` is equal to a `Record<string,string>` today.

Reply:
question: Is a `isRecordStringString` check not too strict? What if the returned interface would look something like:
```ts
{
  status: 123,
  message: 'Lorem',
  other: 'Ipsum',
}
```
`status` being a number would fail the predicate. Do we not just want to check for `isPlainObject` and perhaps that a specific key exists or something similar?

Reply:
I've now made changes to this and now we're using `isPlainObject`.